### PR TITLE
feat(integrations): add support for cal.com

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Add support for cal.com

## Issue ticket number and link
[EXT-130](https://linear.app/nango/issue/EXT-130/add-support-for-calcom)

## Summary by CodeRabbit

- **New Features**
	- Added detailed documentation for integrating with the Cal.com API, including configuration and features.
	- Introduced a new integration card for Cal.com in the productivity tools section.
	- Expanded integration capabilities by adding support for both Cal.com API versions (V1 and V2).

- **Documentation**
	- Enhanced onboarding resources with links to API key generation, authentication, and rate limiting.

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -429,6 +429,34 @@ builder-io-public:
...
+        base_url: https://api.cal.com/v1
+        query:
+            apiKey: ${apiKey}
+        retry:
+            at: 'X-RateLimit-Reset'
+        verification:
+            method: GET
+            endpoint: /me
+    docs: https://docs.nango.dev/integrations/all/cal-com
+cal-com-v2:
```
Comment (bodinsamuel) [inline packages/shared/providers.yaml:446]: feels like we are going to hit more and more of those, I wish we had a better way to version providers

**Review by bodinsamuel — APPROVED** [review]

**Review by TBonnin — APPROVED** [review]
looks good. the logo looks fully transparent to me though

File: packages/shared/providers.yaml (hunk 1 above)
Comment (hassan254-prog) [reply packages/shared/providers.yaml:446]: I agree 💯, just like adding domain extensions to names, e.g., cal.com.

**Comment by hassan254-prog** [comment]
@TBonnin , the Nango black background should look quite nice.
image

**Pull request merged**